custom_components/teslafi/coordinator.py

```diff
@@ -104,6 +104,9 @@ async def _refresh(self) -> TeslaFiVehicle:
             assert last_good.vin
 
         self._vehicle.update_non_empty(current)
+        # Refresh TeslaFi command counts
+        if (counts := current.get("tesla_request_counter", {})):
+            self._vehicle.update_non_empty(counts)
```

Comment:
And we could probably even go one step further and put the `tesla_request_counter` handling directly in `update_non_empty` 🤔 

That would remove the duplication between here and `execute_command`

Reply:
That would look something like this: https://github.com/jhansche/ha-teslafi/blob/main/custom_components/teslafi/model.py#L59-L66
```python
    def update_non_empty(self, data) -> None:
        """Update this object with non-empty data from `data`."""
        if (counts := data.pop("tesla_request_counter", {})):
            data.update(counts)
        if not self.data:
            # Start out with all fields
            super().update(data)
        else:
            filtered = {k: v for (k, v) in data.items() if v}
            super().update(filtered)
```

It also flattens the `tesla_request_counter` key, using `pop()` instead of `get()`